checks/libs/wmi/sampler.py

```diff
@@ -509,12 +509,11 @@ def _parse_results(self, raw_results, includes_qualifiers):
                     wmi_property.Name not in self._property_counter_types
                 )
 
-                if should_get_qualifier_type:
-
-                    # Can't index into "Qualifiers_" for keys that don't exist
-                    # without getting an exception.
-                    qualifiers = dict((q.Name, q.Value) for q in wmi_property.Qualifiers_)
+                # Can't index into "Qualifiers_" for keys that don't exist
+                # without getting an exception.
+                qualifiers = dict((q.Name, q.Value) for q in wmi_property.Qualifiers_)
```

Comment:
If I understand correctly this change makes the logic access the Qualifiers at each iteration, which is costly according to the comment 15 lines above:

> IMPORTANT: To improve performance, only access the Qualifiers if the "CounterType" hasn't already been cached.

If that concern is valid, what do you think about caching the results of the test `CIMTYPE in qualifiers` in a dict on the `sampler` object and access that dict after the first iteration? (having a similar logic as what's done for `CounterType` basically)